### PR TITLE
Add errorhandling for ulimit

### PR DESCRIPTION
**Pull request by glemsom**
In case the container is not spawned with `SYS_RESOURCE`, setting unlimited size for coredumps will never work.
Since the defaults in the script is to abort on any uncaught errors - I propose we catch the `ulimit` error, and notify the user about the possible missing capability.

This fixes issues like https://github.com/zalando/postgres-operator/issues/2387

### CODE REVIEW

File: postgres-appliance/runit/patroni/run [hunk 1]
```diff
@@ -18,7 +18,11 @@ then
 fi
 
 # Enable core dumps
-ulimit -c unlimited
+if ! ulimit -c unlimited
+then
+   echo "Failed to set unlimited size for coredump"
+   echo "Consider adding the SYS_RESOURCE capability"
```
Comment (hughcapet) [inline postgres-appliance/runit/patroni/run:24]:
```suggestion
```

**Comment by hughcapet** [comment]
I would abstain from adding any specific configuration suggestions to the logging here but let's add the error handling anyway. Thanks.

**Comment by idanovinda** [comment]
👍

**Comment by hughcapet** [comment]
👍

**Pull request merged**